do/common.go

```diff
@@ -0,0 +1,53 @@
+package do
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/digitalocean/godo"
+	"github.com/digitalocean/godo/context"
+)
+
+const apiPerPage = 100
+
+func allDropletList(ctx context.Context, client *godo.Client) ([]godo.Droplet, error) {
+	// create a list to hold our droplets
+	list := []godo.Droplet{}
+
+	// create options. initially, these will be blank
+	opt := &godo.ListOptions{PerPage: apiPerPage}
+	for {
+		droplets, resp, err := client.Droplets.List(ctx, opt)
+		if err != nil {
+			return nil, err
+		}
+
+		if resp == nil {
+			return nil, fmt.Errorf("DO API returned no response ")
+		}
+
+		if resp.StatusCode != http.StatusOK {
```

Comment:
Probably this one is not needed.
`godo` already checks for requests outcome, and returns an error if it's outside the 200-299 range, so your first error check above would catch it.

https://github.com/digitalocean/godo/blob/master/godo.go#L316